contracts/contracts/processors/BaseProcessor.sol

```diff
@@ -68,4 +68,29 @@ contract BaseProcessor is Ownable {
         require(!nullifierRegistry.isNullified(_nullifier), "Nullifier has already been used");
         nullifierRegistry.addNullifier(_nullifier);
     }
+
+    // stringToUint is used to convert a string like "45" to a uint256 4
+    function _stringToUint(string memory s, uint256 expectedDecimals) internal pure returns (uint256) {
+        bytes memory b = bytes(s);
+        uint256 result = 0;
+        bool decimals = false;
+        uint256 decimalPlaces;
+        for (uint256 i = 0; i < b.length; i++) {
+            if (b[i] >= 0x30 && b[i] <= 0x39) {
+                result = result * 10 + (uint256(uint8(b[i])) - 48);
+            }
+
+            if (decimals) {
+                decimalPlaces++;
```

Comment:
Think this implementation has a bug. 

I'm assuming, by expected decimals, you mean number of digits towards the right of decimal point. If that is the case, then why are we incrementing decimals counter, after the decimal point has been found. Shouldn't we keep incrementing before and stop incrementing once we found the decimal point going from left?

Consider the example of 999.0, with expected decimal points 2, `decimalPlaces` will be 4. And this function reverts. Instead, it should return, `99900`.

Reply:
It would also revert, for correct cases. For example, when the amount is 999.00, expected decimal points is 2, but decimalPlaces will be 3.

Reply:
Think, I got it wrong, from the variable names, `expectedDecimals` is apparently 6 instead of 2. 
`amount = _stringToUint(_parseSignalArray(_proof.signals, 4, 6), 6);`
Can you add docs to this function?